caPutLogApp/caPutLogAs.c

```diff
@@ -140,9 +151,17 @@ static void caPutLogAs(asTrapWriteMessage *pmessage, int afterPut)
         plogData->pfield = paddr->pfield;
 
         options = 0;
-        num_elm = 1;
+        num_elm = caPutLogMaxArraySize(plogData->type);
         status = dbGetField(
             paddr, plogData->type, &plogData->old_value, &options, &num_elm, 0);
+        plogData->old_log_size = num_elm;
+        if ((prset = dbGetRset(paddr)) &&
+                prset->get_array_info) {
+            prset->get_array_info(paddr, &noElements, &offset);
+            plogData->old_size = noElements;
+        } else {
+            plogData->old_size = num_elm;
+        }
```

Comment:
I changed the logic how is determined if a value is an array here: b5a8e866b446674a4bdde15b328752ac3a5a10c9

I extracted it to a separate function instead of putting it before the `if` statement, as that would require more changes to existing code (most pointers with required information are obtained inside the `if` statement ).